ui/apps/platform/src/services/ComplianceResultsService.ts

```diff
@@ -85,11 +87,24 @@ export function getComplianceProfilesStats(): Promise<ListComplianceProfileScanS
  * Fetches the profile check results.
  */
 export function getComplianceProfileResults(
-    profileName: string
+    profileName: string,
+    page: number,
+    pageSize: number
 ): Promise<ListComplianceProfileResults> {
+    let offset: number | undefined;
+    if (typeof page === 'number' && typeof pageSize === 'number') {
+        offset = page > 0 ? page * pageSize : 0;
+    }
+    const queryParameters = {
+        query: {
+            pagination: { offset, limit: pageSize },
+        },
+    };
+    const params = qs.stringify(queryParameters, { arrayFormat: 'repeat', allowDots: true });
```

Comment:
We can create a util function for generating the query parameters (with pagination, search, and sorting) if we don't already have one. I imagine we'll need to do this for all the tables in Compliance 2.0.

Reply:
I'm open to that. Mark pointed out the `getPaginationParams` util function which makes sense. I've thought about doing something like that but I wonder if further abstraction would be more or less beneficial. I'm hoping we'll get a better idea once we add search filters and more services.

Reply:
I think it would be beneficial. I've noticed a lot of copying logic when it comes to constructing the pagination query and rendering the pagination component (from my experience, seems like Dave also agreed). Might be a good opportunity to see if we can simplify things. You can leave things for now. When we need to add it for the next table, let's see if we can make it then.